dora/src/bytecode/generator.rs

```diff
@@ -437,18 +428,23 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
             Register::zero()
         };
         self.gen.set_position(expr.pos);
-        let arg_start_reg = match *call_type {
+        let (arg_start_reg, nil_check) = match *call_type {
```

Comment:
I think you can move that `nil_check` variable down into the next match for `CallType::Method`.

Reply:
Yep, will change it, when I come home.